### PR TITLE
Various adjustments to the WebLogic rules.

### PR DESCRIPTION
**Pull request by Maarc**
Removed classifications leading to duplicated entries. Replaced ignore-references by a catchall. Adjusted titles. Updated effort for the WebLogic catchall.

### CODE REVIEW

**Comment by windupgithubbot1** [comment]
Can one of the admins verify this patch?

File: rules-reviewed/eap6/weblogic/weblogic-ignore-references.windup.xml [hunk 1]
```diff
@@ -1,31 +0,0 @@
...
-         <javaclass-ignore reference-prefix="weblogic.ejb.container.persistence.spi.EloWrapper"/>
-         <javaclass-ignore reference-prefix="weblogic.ejb20.interfaces"/>
-         <javaclass-ignore reference-prefix="weblogic.ejb20.internal"/>
-         <javaclass-ignore reference-prefix="weblogic.ejbgen"/>
-         <javaclass-ignore reference-prefix="weblogic.ejb20.persistence.spi.CMPBean"/>
-         <javaclass-ignore reference-prefix="weblogic.ejb20.persistence.spi.EoWrapper"/>
-         <javaclass-ignore reference-prefix="weblogic.wsee.tools.clientgen"/>
-         <javaclass-ignore reference-prefix="weblogic.rmi.internal"/>
-         <javaclass-ignore reference-prefix="weblogic.utils.PlatformConstants"/>
-         <javaclass-ignore reference-prefix="weblogic.webservice.core.rpc"/>
```
Comment (mareknovotny) [inline rules-reviewed/eap6/weblogic/weblogic-ignore-references.windup.xml]: @Maarc  what about these (`weblogic.[wsee|rmi|utils|webservice]`) you didn't add in **weblogic-ejb-01000** rule?

File: rules-reviewed/eap6/weblogic/weblogic-webapp.windup.xml [hunk 2]
```diff
@@ -100,7 +100,7 @@
                       See the [javax.servlet.annotation JavaDoc](http://docs.oracle.com/javaee/6/api/javax/servlet/annotation/package-summary.html) for more information.
                       ]]>
                     </message>
-                    <link href="https://access.redhat.com/articles/1249423" title="Migrate WebLogic Proprietary Servlet Annotations" />
+                    <link href="https://access.redhat.com/articles/1249423" title="WebLogic proprietary servlet annotations" />
```
Comment (mareknovotny) [inline rules-reviewed/eap6/weblogic/weblogic-webapp.windup.xml]: About changed titles for links - These are trying to match titles in original linked documents, rephrasing them to something different could be confusing for a reader.

File: rules-reviewed/eap6/weblogic/weblogic-ejb.windup.xml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        <dependencies>
+            <addon id="org.jboss.windup.rules,windup-rules-javaee,2.3.0.Final" />
+            <addon id="org.jboss.windup.rules,windup-rules-java,2.3.0.Final" />
+            <addon id="org.jboss.windup.rules,windup-rules-xml,2.3.0.Final" />
+        </dependencies>
+        <sourceTechnology id="weblogic" />
+        <targetTechnology id="eap" versionRange="[6,)" />
+    </metadata>
+    <rules>
+        <rule id="weblogic-ejb-01000">
```
Comment (mareknovotny) [inline rules-reviewed/eap6/weblogic/weblogic-ejb.windup.xml]: I don't see any test for this new ruleset

File: rules-reviewed/eap6/weblogic/weblogic-webapp.windup.xml (hunk 2 above)
Comment (Maarc) [inline rules-reviewed/eap6/weblogic/weblogic-webapp.windup.xml]: Ok! It is fine to keep them in general. Sometimes the original titles are quite terrible or have Too Many Words Starting With A Capital Letter ;-)

File: rules-reviewed/eap6/weblogic/weblogic-ejb.windup.xml (hunk 3 above)
Comment (Maarc) [inline rules-reviewed/eap6/weblogic/weblogic-ejb.windup.xml]: Indeed. I have not provided any.

File: rules-reviewed/eap6/weblogic/weblogic-ignore-references.windup.xml (hunk 1 above)
Comment (Maarc) [inline rules-reviewed/eap6/weblogic/weblogic-ignore-references.windup.xml]: Indeed, I intentionally removed this "ignore WebLogic references" ruleset. 

(weblogic.[wsee|rmi|utils|webservice]) would land in the potential issue bucket. Ideally, in the future, separate rules should be written for each pattern. Those patterns definitely cannot be ignored.

**Comment by Maarc** [comment]
Is anything missing to merge this PR?

**Comment by mareknovotny** [comment]
@Maarc yes, there are missing things. If you go through the comments from past there are required changes to do. Therefore this PR is marked with label as **Reviewed/Needs Adjustment**

Sum up:
* missing tests -> https://github.com/windup/windup-rulesets/pull/188#discussion_r96389180
* titles for linked documents are trying to match the originals - you changed them -> https://github.com/windup/windup-rulesets/pull/188#discussion_r96388888
* dropped ignores which is not redundant.

**Comment by OndraZizka** [comment]
Marc, do you think you have time to update the PR, or should we update them?

File: rules-reviewed/eap6/weblogic/weblogic-ignore-references.windup.xml (hunk 1 above)
Comment (OndraZizka) [inline rules-reviewed/eap6/weblogic/weblogic-ignore-references.windup.xml]: Marc, why did yo uremove them? I agree that Windup should not just chicken out when seeing these and fight :) But I assume there was a reason for it. I assume these interfaces are build-time libraries and internal IBM server libs. So should they be reported as "trash this"?

**Comment by OndraZizka** [comment]
@Maarc  see https://github.com/Maarc/windup-rulesets/pull/1
Although it looks there is a couple of recent PRs somehow tangled into that.

**Comment by OndraZizka** [comment]
Continued in #202. Closing.

File: rules-reviewed/eap6/weblogic/weblogic-ignore-references.windup.xml (hunk 1 above)
Comment (mareknovotny) [inline rules-reviewed/eap6/weblogic/weblogic-ignore-references.windup.xml]: One additional thing to these ignores, we added them mostly that we got or know that those packages are used for automatically generated stub classes. If they are not correct please link some docs too as we marked and ignored them for this specific reason. Generated classes are waste of time for decompilation and analysis.

Comment (jsight) [inline rules-reviewed/eap6/weblogic/weblogic-ignore-references.windup.xml]: Yeah, these were brought over from Windup 0.6 to improve performance as they had a big impact in the past.

**Pull request closed without merging**